public/app/features/dashboard/api/ResponseTransformers.ts

```diff
@@ -293,14 +299,29 @@ function getElementsFromPanels(panels: Panel[]): [DashboardV2Spec['elements'], D
 function getDefaultDatasourceType() {
   const datasources = config.datasources;
   // find default datasource in datasources
-  return Object.values(datasources).find((ds) => ds.isDefault)!.type;
+  const dsList = Object.values(datasources);
+  let dsType = dsList.find((ds) => ds.isDefault)?.type;
+  if (!dsType) {
+    const dsRefFromBoot = getDefaultDataSourceRef();
+    dsType = dsRefFromBoot?.type ?? 'grafana'; // TODO: is there any other type of datasource type?
+  }
+
+  return dsType;
 }
 
 function getDefaultDatasource(): DataSourceRef {
   const datasources = config.datasources;
 
-  // find default datasource in datasources
-  const defaultDs = Object.values(datasources).find((ds) => ds.isDefault)!;
+  // find default datasource in datasources from config using the isDefault flag
+  let defaultDs = Object.values(datasources).find((ds) => ds.isDefault)!;
+
+  if (!defaultDs) {
+    // we also can get default datasource from config.bootData.settings.defaultDatasource;
+    const dsRefFromBoot = getDefaultDataSourceRef();
+    if (dsRefFromBoot) {
+      defaultDs = Object.values(datasources).find((ds) => ds.name === dsRefFromBoot.uid)!;
+    }
+  }
```

Comment:
Same here - i thnk `getDefaultDataSourceRef` is sufficient to retreive the uid and do the rest